### PR TITLE
Add software description to setup.py

### PR DESCRIPTION
**Pull request by okuta**
This PR provides rich information for [PyPI](https://pypi.org/project/cupy/).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -137,6 +137,27 @@ def reduce_requirements(key):
...
 exec(open(os.path.join(here, 'cupy', '_version.py')).read())
 
+CLASSIFIERS = """\
+Development Status :: 5 - Production/Stable
+Intended Audience :: Science/Research
+Intended Audience :: Developers
+License :: OSI Approved :: MIT License
+Programming Language :: Python
+Programming Language :: Python :: 2
+Programming Language :: Python :: 2.7
```
Comment (leofang) [inline setup.py]: Shouldn't you add these two lines only when backporting to v6? In master/v7 the support for PY2 is dropped.

File: setup.py [hunk 2]
```diff
@@ -146,6 +167,13 @@ def reduce_requirements(key):
     author_email='tokui@preferred.jp',
     url='https://docs-cupy.chainer.org/',
     license='MIT License',
+    download_url="https://https://pypi.org/project/cupy/",
```
Comment (toslunar) [inline setup.py]:
```suggestion
    download_url="https://pypi.org/project/cupy/",
```

**Comment by okuta** [comment]
I fixed.

File: setup.py (hunk 1 above)
Comment (okuta) [reply setup.py]: Thanks, I fixed.

File: setup.py [hunk 3]
```diff
@@ -146,6 +166,13 @@ def reduce_requirements(key):
     author_email='tokui@preferred.jp',
     url='https://docs-cupy.chainer.org/',
     license='MIT License',
+    download_url="https://pypi.org/project/cupy/",
```
Comment (kmaehashi) [inline setup.py]: It seems that other projects use this field as a link to tarballs.
https://sherif.io/2016/09/30/Python-package-with-GitHub-PyPI.html#add-tarball-download-url-to-setuppy

But it's not required and is not used even in pypa's official [example project](https://github.com/pypa/sampleproject/blob/master/setup.py). So how about removing unless if you need some reason to include this?

File: setup.py (hunk 1 above)
Comment (kmaehashi) [reply setup.py]: Thanks, I'll add it in the backport.

File: setup.py (hunk 3 above)
Comment (okuta) [reply setup.py]: OK, I will remove this line.

**Comment by okuta** [comment]
I fixed.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Pull request merged**